lib/card.rb

```diff
@@ -0,0 +1,18 @@
+class Card
+
+  attr_reader :id
+
+  def initialize(id, attributes)
+    @id = id
+    @attributes = attributes
+  end
+
+  def method_missing(method_name, *args)
```

Comment:
I think this is a decent approach here. Only in rare cases will this stop the regular flow, you run into issues when this starts doing unexpected things, but this seems like a decently constrained use case.

Another more explicit option would be something like this:
```ruby
def initialize(id, attributes)
  @id = id
  @attributes = attributes
  define_attribute_methods
end

def define_attribute_methods
  @attributes.keys.each do |key|
    define_method key do
      @attributes[key]
    end
  end
end
```